### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by shafiuyushawu**
### Hello Reviewer 👍 
This is a pull request to create `OOP School Liabrary` setting up these classes `nameable`, `base_decorator`, `capitalize_decorator`, `trimmer_decorator` and modified `person`. Below is the highlights.

# **Functionality implemented ✍️**
- Created class `nameable` implemented a method called `correct_name` that will raise a `NotImplementedError` ✔️ 
- Created class `base_decorator` that `inherent` from `nameable` with constructor with a method `correct_name`✔️ 
- Created class `capitalize_decorator` that inherent from`base_decorator` with a method `correct_name` which action is conver name to initial capital ✔️ 

- Created class `trimmer_decorator` that inherent from `base_decorator` with a method `correct_name` that returns the name with its `chars` less than 10 ✔️

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
```
Comment (Zuhaib042) [inline base_decorator.rb]: - [x] I have ran your code and it gives error. but if you remove `super` keyword from line 5 and add `attr_accessor :nameable` right after line 3, then you code will be working. kindly consider fixing the bug.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @shafiuyushawu 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Implemented most of the requirements

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (shafiuyushawu) [reply base_decorator.rb]: Hello @Zuhaib042,
The bug is resolved, thanks for your insight, much appreciates.

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @shafiuyushawu :wave: 
 
Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark: Descriptive pull request.
:heavy_check_mark: Correct Git flow.
:heavy_check_mark: Implement all the requirements.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by shafiuyushawu** [comment]
Hello @Tufahel

Thank you!

**Pull request merged**